apps/user/loop.c

```diff
@@ -0,0 +1,27 @@
+#include "string.h"
+#include "app.h"
+#include "egos.h"
+
+void loop(int count) {
+    int counter = 0;
+    for(int i=0;i<count*QUANTUM; i++) {
+        float ret = (float)i / counter;
+        counter++;
+    }
+}
+
+void sleep(int time) {
+  /* Student code goes here (Premptive scheduler)
+   * call proc_sleep to sleep for time * QUANTUM.
+   * */
+}
+
+int main(int argc, char** argv) {
+    if (argc == 1) {
+        loop(100);
+    } else if (strcmp(argv[1], "sleep") == 0) {
+        INFO("Sleep for 10 QUANTUM...\n");
+        sleep(10);
+        INFO("                   ...and waked up\n");
+    }
+}
```

Comment:
Given that we can now install apps easily, let's keep this file somewhere else. In the project handout, we can ask students to install this app.

In the EGOS book, I can put this file in the book Github repo.